internal/logging/rotate.go

```diff
@@ -39,3 +45,46 @@ func rotateLogs(files []fs.FileInfo, timeCutoff time.Time, amountCutoff int) []f
 
 	return rotate
 }
+
+func rotateLogsOnDisk() {
+	// Clean up old log files
+	logDir := filepath.Dir(FilePath())
+	files, err := ioutil.ReadDir(logDir)
+	if err != nil && !os.IsNotExist(err) {
+		Error("Could not scan config dir to clean up stale logs: %v", err)
+		return
+	}
+
+	// Prevent running over this logic too often as it affects performance
+	// https://activestatef.atlassian.net/browse/DX-1516
+	if len(files) < 30 {
+		return
+	}
+
+	rotate := rotateLogs(files, time.Now().Add(-time.Hour), 10)
+	for _, file := range rotate {
+		if err := os.Remove(filepath.Join(logDir, file.Name())); err != nil {
+			Error("Could not clean up old log: %s, error: %v", file.Name(), err)
+		}
+	}
+}
+
+// RotateLogListener rotates logs on a timer.
+// Run this as a Go routine.
+func RotateLogTimer() {
+	timeout := 1 * time.Minute
+	if durationString := os.Getenv(constants.SvcLogRotateTimerDurationEnvVarName); durationString != "" {
+		if duration, err := strconv.Atoi(durationString); err == nil {
+			timeout = time.Duration(duration) * time.Second
+		}
+	}
+
+	rotateLogsOnDisk()
+	for {
+		tick := time.Tick(timeout)
+		select {
+		case <-tick:
+			rotateLogsOnDisk()
+		}
```

Comment:
Please consider using `time.After`. Also consider the note in the docs regarding cleanup (I think we should be good to go using After, but give it your own judgment): https://pkg.go.dev/time#After